vllm/v1/sample/ops/topk_topp_sampler.py

```diff
@@ -0,0 +1,156 @@
+from typing import Dict
+
+import torch
+import torch.nn as nn
+
+from vllm.logger import init_logger
+from vllm.platforms import current_platform
+
+logger = init_logger(__name__)
+
+try:
+    import flashinfer.sampling
+    use_flashinfer = True
+except ImportError:
+    use_flashinfer = False
+
+
+class TopKTopPSampler(nn.Module):
+
+    def __init__(self):
+        super().__init__()
+        if current_platform.is_cuda:
+            if use_flashinfer:
+                self.forward = self.forward_cuda
+            else:
+                logger.warning(
+                    "FlashInfer is not available. Falling back to the PyTorch-"
+                    "native implementation of top-p & top-k sampling. For the "
+                    "best performance, please install FalshInfer.")
+                self.forward = self.forward_native
```

Comment:
IIRC previously the FlashInfer sampler outputs different results as the native one, so we make it configurable even FlashInfer is installed, but I'm not sure if this is still the case. cc @yzh119.

Reply:
Yeah we actually have `VLLM_USE_FLASHINFER_SAMPLER` env variable which is False by default.

I do want users to benefit from this by default though, given the considerable perf difference. Can we enable this env variable by default? Otherwise, I would personally prefer to leave this as is (v1 using it whenever available).

Reply:
Hmm can we set default to True for v1 only?

Reply:
@comaniac Good point. I made VLLM_USE_FLASHINFER_SAMPLER None by default and interpret it as True only for V1.

Reply:
Thanks! Looks all good to me